src/index.js

```diff
@@ -213,16 +219,137 @@ module.exports = class IPLDResolver {
 
       pull(
         pull.values([nodeAndCID]),
-        this._putStream(callback)
+        this._putStream((err) => {
+          if (err) {
+            return callback(err)
+          }
+          callback(null, nodeAndCID.cid)
+        })
+      )
+    }
+  }
+
+  treeStream (cid, path, options) {
+    if (typeof path === 'object') {
+      options = path
+      path = undefined
+    }
+
+    options = options || {}
+
+    // non recursive
+    const p = pullPushable()
```

Comment:
why do you need to use a push stream? you should be able to return source stream that only walks the next thing if it actually needs to